magento/pwa-studio
# fix(install): Use npm-is to require Yarn before install. Fixes PWA-505.

## Description

Add a preinstall script that prevents `npm install`, requiring `yarn install` instead.

The monorepo won't work right if installed with `npm`. Several users have reported that `npm install` at repo root causes somewhat mysterious errors. This is because the repo uses Yarn Workspaces, and Yarn automatically installs workspace dependencies along with root dependencies. NPM doesn't, so packages are missing deps when they run.

I, uh, found a utility called [`npm-is`](https://www.npmjs.com/package/npm-is) which detects and asserts package managers, and used it to add a preinstall script `npx npm-is yarn`. This command fails if the NPM script wasn't invoked with Yarn, thus preventing `npm install`.

## Related Issue
Closes PWA-505.

## Acceptance 
### Verification Stakeholders
@dpatil-magento 

### Verification Steps
1. Run `yarn run clean:all` to remove all node_modules folders.
2. Run `npm install` at package root.
3. Observe that it fails before installing dependencies, throwing an informative error message.

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
